### PR TITLE
Added stream option to docker.compose.run and fixed the tty flag

### LINKED ISSUE #248: compose run
Hello,
I know that a number of compose commands are still not implemented and I can imagine that you have your own priority list about them, but in case of tie can I ask you to push the run command higher in the ranking? :smile: 

I think that compose run is the last command that I need to complete the migration of all my use cases to python on whales, so the implementation of that command would be very appealing to me, but as usual I don't want to bother you with many requests, also considering that I have a number of other issues still open... so please consider this as a low priority request

Thank you again!

### PR DESCRIPTION
**Pull request by mdantonio**
Fixes #248

### CODE REVIEW

**Comment by mdantonio** [comment]
Due to the change from `run(full_cmd)` to `run(full_cmd, tty=tty)` the following test started to fail:

```
def test_compose_run_detach():
    container = docker.compose.run("alpine", ["echo", "dodo"], detach=True)

    time.sleep(0.1)
    assert not container.state.running
    assert container.logs() == "dodo\n"
```

This is because the run wrapper with tty enabled directs the output to sys.stdout/err instead of subprocess.PIPE, so that the test is unable to obtain the container name

Disabled the tty ( => `container = docker.compose.run("alpine", ["echo", "dodo"], detach=True, tty=False)`) to fix the test, I hope that it is the correct approach

File: python_on_whales/components/compose/cli_wrapper.py [hunk 1]
```diff
@@ -252,7 +253,41 @@ def run(
...
+            raise ValueError(
+                "You can't set tty=True and stream=True at the same"
+                "time. Their purpose are not compatible."
+            )
+
+        if detach and stream:
+            raise ValueError(
+                "You can't detach and stream at the same time. It's not compatible."
+            )
+
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/compose/cli_wrapper.py:290]: I would also add here this:

```suggestion
            )

    if detach and tty:
            raise ValueError(
                "You can't detach and set tty=True at the same time. It's not compatible. "
                "Try setting tty=False in docker.compose.run(...)."
            )
```

File: python_on_whales/components/compose/cli_wrapper.py [hunk 2]
```diff
@@ -252,7 +253,41 @@ def run(
...
+        # Returns:
+            Optional[str]
+
+        """
+
+        if tty and stream:
+            raise ValueError(
+                "You can't set tty=True and stream=True at the same"
+                "time. Their purpose are not compatible."
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/compose/cli_wrapper.py:283]:
```suggestion
                "time. Their purpose are not compatible. Try setting tty=False in docker.compose.run"
```

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]
A few changes to be made, but it's very nice overall :) thanks for the pull request!

**Review by gabrieldemarmiesse — APPROVED** [review]

**Pull request merged**